### PR TITLE
fix: deleted actions display error messages when navigated to

### LINKED ISSUE #9485: [Bug]: Opening URL of a deleted Google Sheets action shows a faux blank UI
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When opening the URL of a Google Sheets query that's already deleted, we expect to see a "not-found" or 404 kinda page. Instead we see a blank query form. This is very misleading.

### Steps To Reproduce

1. Connect to Google Sheets and create a new query to perform some operation on a Google Sheet.
2. Now open the URL for this sheet in a new tab.
3. Delete this action in tab 1.
4. Go to tab 2 and refresh to see the problem.

Video demonstrating the problem:

https://user-images.githubusercontent.com/120119/144195506-7c8df7c9-36bc-44e7-a459-19a241fdd545.mov

### Environment

Production

### Version

Cloud

### PR DESCRIPTION
**Pull request by Irongade**
## Description
Better error handling for deleted actions, when Users navigate to a deleted action, they are shown an error and prompted to go back.

Fixes #9485.

- Bug fix (non-breaking change which fixes an issue).

Manual Testing employed.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: fix/deleted-action-error-handling 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 55.13 **(-0.02)** | 37.06 **(-0.03)** | 33.96 **(-0.01)** | 55.64 **(-0.01)**
 :red_circle: | app/client/src/pages/Editor/EntityNotFoundPane.tsx | 72.73 **(0)** | 50 **(-50)** | 0 **(0)** | 72.73 **(0)**
 :red_circle: | app/client/src/pages/Editor/QueryEditor/index.tsx | 30.56 **(-1.44)** | 8.16 **(-1.14)** | 5.88 **(-0.37)** | 31.68 **(-1.3)**
 :red_circle: | app/client/src/pages/Editor/SaaSEditor/ActionForm.tsx | 26.39 **(-0.88)** | 0 **(0)** | 0 **(0)** | 28.79 **(-1.21)**
 :red_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.71 **(-0.23)** | 40.83 **(-0.84)** | 36.21 **(0)** | 56.74 **(-0.25)**</details>

### CODE REVIEW

File: app/client/src/pages/Editor/EntityNotFoundPane.tsx [hunk 1]
```diff
@@ -46,7 +46,7 @@ const Wrapper = styled.div`
   }
 `;
 
-function EntityNotFoundPane() {
+function EntityNotFoundPane(props: any) {
```
Comment (rashmi-rai) [inline app/client/src/pages/Editor/EntityNotFoundPane.tsx:49]: Can we add type here?

Comment (Irongade) [reply app/client/src/pages/Editor/EntityNotFoundPane.tsx]: Done @AmanAgarwal041  @rashmi-rai

**Review by rashmi-rai — DISMISSED** [review]

**Review by AmanAgarwal041 — APPROVED** [review]
LGTM

**Review by rashmi-rai — APPROVED** [review]

**Review by ayushpahwa — APPROVED** [review]

**Review by AmanAgarwal041 — APPROVED** [review]

**Pull request merged**